src/main/scala/fm/xml/XmlReader.scala

```diff
@@ -64,143 +101,186 @@ object XmlReader {
-  
-  private[this] val itemClass: Class[T] = classTag[T].runtimeClass.asInstanceOf[Class[T]]
+  *
+  * @param itemPath The XPath-like path to the element we are interested in.  (e.g. "part", "items/part", etc)
+  * @param toMappedValue Map the XmlValue into another return type, this should match the return type in MultiXmlReader
+  * @tparam XmlValue The unmarshalled element class type
+  * @tparam MappedXmlValue A mapped XmlValue type
+  */
+final case class XmlReaderPath[
+XmlValue: ClassTag,
+MappedXmlValue: ClassTag,
+](itemPath: String, toMappedValue: XmlValue => MappedXmlValue) extends Logging {
+  private[this] val itemClass: Class[XmlValue] = classTag[XmlValue].runtimeClass.asInstanceOf[Class[XmlValue]]
   private[this] val jaxbContext: JAXBContext = JAXBContext.newInstance(itemClass)
-  
+  private[this] val unmarshaller: Unmarshaller = jaxbContext.createUnmarshaller()
+
   // The XPath-like path to the element we are interested in
   // part => Array("part"), items/part => Array("items","part")
-  private[this] val path: Array[String] = itemPath.split('/')
-  
+  private[xml] def path: Array[String] = itemPath.split('/')
+
   // The name of the element we care about (last part of the path)
   // items/part => part
-  private[this] val itemName: String = path.last
-  
-  private[this] val targetDepth: Int = path.length - 1
-  
+  private[xml] def itemName: String = path.last
+
+  private[xml] def targetDepth: Int = path.length - 1
+
+  private[xml] def readValue(xmlStreamReader: XMLStreamReader2): MappedXmlValue =  {
+    xmlStreamReader.require(START_ELEMENT, null, itemName)
+
+    val value: XmlValue = unmarshaller.unmarshal(xmlStreamReader, itemClass).getValue
+    toMappedValue(value)
+  }
+
+  // TODO: Cleanup the matches/pathMatches
+  // Check to see if the target depth and path match
+  private[xml] def matches(xmlStreamReader: XMLStreamReader2, currentDepth: Int, currentPath: Array[String]): Boolean = {
+    targetDepth == currentDepth &&
+      xmlStreamReader.getLocalName === itemName &&
+      pathMatches(xmlStreamReader, currentDepth, currentPath)
+  }
+
+  private[xml] def pathMatches(xmlStreamReader: XMLStreamReader2, currentDepth: Int, currentPath: Array[String]): Boolean = {
+    // Make sure previous paths match this & make sure the next item is either the next in the path sequence, or the target itemName
+    val previousPathMatches: Boolean = if (currentDepth > 0) (0 until currentDepth).forall{ depth: Int => currentPath(depth) === path(depth) } else true
+    val currentPathMatches: Boolean = xmlStreamReader.getLocalName === path(currentDepth) || xmlStreamReader.getLocalName === itemName
+
+    previousPathMatches && currentPathMatches
+  }
+}
+
+
+/**
+  * defaultNamespaceURI - If no namespace is set, use this as a default
+  * overrideDefaultNamespaceURI - Override the namespace and just use this instead
+  */
+class XmlReader[T: ClassTag](
```

Comment:
Why is `: ClassTag` being used here?

Reply:
It has to do with the `targets: Seq[XmlReaderPath[_, T]]` signature which actually does the jaxb unmarshalling, I *believe* there was a classTag requirement for that code block...this then creates a sequence of things that need the `: ClassTag`

Reply:
Not sure what you are trying to say here...  The `T` here has no involvement in jaxb unmarshalling.

Reply:
I'll take a look, I assumed the symmetry for the types was required for the eventual use in the element unmarshalling...

 ```
  private[this] val itemClass: Class[XmlValue] = classTag[XmlValue].runtimeClass.asInstanceOf[Class[XmlValue]]
  private[this] val jaxbContext: JAXBContext = JAXBContext.newInstance(itemClass)
  private[this] val unmarshaller: Unmarshaller = jaxbContext.createUnmarshaller()
```